system/metrics/sinks/file/fileSink.hpp

```diff
@@ -0,0 +1,51 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#pragma once
+
+#include "jmetrics.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+#include "jfile.ipp"
+#include "jsem.hpp"
+#include <thread>
+#include <condition_variable>
+
+
+using namespace hpccMetrics;
+
+class DECL_EXPORT FileMetricSink : public MetricSink
```

Comment:
This should be based on a conditional define.  You had code previously to implement it, but I commented that the name was verbose/did not match the general pattern used elsewhere.